internal/pkg/agent/application/dispatcher/dispatcher.go

```diff
@@ -146,14 +147,18 @@ func (ad *ActionDispatcher) Dispatch(ctx context.Context, acker acker.Acker, act
 				continue
 			}
 			ad.log.Debugf("Failed to dispatch action '%+v', error: %+v", action, err)
-			ad.errCh <- err
+			reportedErr = err
 			continue
 		}
 		ad.log.Debugf("Successfully dispatched action: '%+v'", action)
 	}
 
 	if err = acker.Commit(ctx); err != nil {
-		ad.errCh <- err
+		reportedErr = err
```

Comment:
Do we need to take the most recent error, or can we just write to the error channel twice? You'd probably need a separate error variable to capture the commit error as well.

Reply:
we can write to error channel twice but we don't have a tracking about what is the outcome of an action. 
if we write directly to a channel it will be picked up by error reporter but we would still need to know if there was something reported to reset it if it was not.
if we also write multiple time, only the last one will be shown in the state. so it does not really make sense to send multiple times and wait for channel read

Reply:
This code is making an assumption about how the error channel will be used by the caller in its implementation right now, but I don't think this matters that much. For now it makes no functional difference.